tensorboard/backend/event_processing/data_provider.py

```diff
@@ -265,29 +265,14 @@ def read_blob_sequences(
             result[run] = result_for_run
             for (tag, metadata) in six.iteritems(tags_for_run):
                 events = self._multiplexer.Tensors(run, tag)
-                data = []
+                data_by_step = {}
                 for event in events:
-                    num_blobs = _tensor_size(event.tensor_proto)
-                    values = tuple(
-                        provider.BlobReference(
-                            _encode_blob_key(
-                                experiment_id,
-                                plugin_name,
-                                run,
-                                tag,
-                                event.step,
-                                i,
-                            )
-                        )
-                        for i in range(num_blobs)
-                    )
-                    data.append(
-                        provider.BlobSequenceDatum(
-                            wall_time=event.wall_time,
-                            step=event.step,
-                            values=values,
-                        )
+                    if event.step in data_by_step:
+                        continue
+                    data_by_step[event.step] = _convert_blob_sequence_event(
+                        experiment_id, plugin_name, run, tag, event
                     )
+                data = [datum for (step, datum) in sorted(data_by_step.items())]
```

Comment:
Just to confirm, this is new now that the results will be sorted by step?  I think that should be fine (and it's consistent with other DataProvider impls) but it is different from before, right?

Reply:
The spec requires this (because it needs to be done at some level and
can usually be done efficiently at the database layer); it’s true that
the earlier draft didn’t ensure that, but that was an oversight.